Userland/Libraries/LibJIT/Assembler.h

```diff
@@ -306,23 +418,21 @@ struct Assembler {
 
     void cmp(Operand lhs, Operand rhs)
     {
-        if (rhs.type == Operand::Type::Imm && rhs.offset_or_immediate == 0) {
+        if (lhs.type == Operand::Type::Reg && rhs.type == Operand::Type::Imm && rhs.offset_or_immediate == 0) {
             test(lhs, lhs);
-        } else if (lhs.type == Operand::Type::Reg && rhs.type == Operand::Type::Reg) {
-            emit8(0x48
-                | ((to_underlying(rhs.reg) >= 8) ? 1 << 2 : 0)
-                | ((to_underlying(lhs.reg) >= 8) ? 1 << 0 : 0));
+        } else if (lhs.is_register_or_memory() && rhs.type == Operand::Type::Reg) {
+            REX::emit_for_mr(*this, lhs, rhs, true);
             emit8(0x39);
-            emit8(0xc0 | (encode_reg(rhs.reg) << 3) | encode_reg(lhs.reg));
-        } else if (lhs.type == Operand::Type::Reg && rhs.type == Operand::Type::Imm && rhs.fits_in_i8()) {
-            emit8(0x48 | ((to_underlying(lhs.reg) >= 8) ? 1 << 0 : 0));
+            ModRM::emit_mr(*this, lhs, rhs);
+        } else if (lhs.is_register_or_memory() && rhs.type == Operand::Type::Imm && rhs.fits_in_i8()) {
+            REX::emit_for_slash(*this, lhs, true);
```

Comment:
This feels very clunky, do we have to say `REX::` and `*this`?
`emit_rex_for_slash(lhs, true)`?

Also, what is `true`? Can we use an enum to avoid the mysterious bool parameter. (This same question for other uses of bool parameters :)

Reply:
I can make them more standalone, lets see if that feels better

Reply:
And the bool parameter is for the W flag in the prefix, which might or might not mean "width override"